conbench/entities/benchmark_result.py

```diff
@@ -507,7 +459,115 @@ def ui_hardware_short(self) -> str:
         return f"{hw.id[:4]}: " + hw.name
 
 
-def do_iteration_samples_look_like_error(data: list[Optional[Decimal]]) -> bool:
+def validate_and_aggregate_samples(stats_usergiven: Any):
+    """
+    Raises BenchmarkResultValidationError upon logical inconsistencies.
+
+    Only run this for the 'success' case, i.e. when the input is a list longer
+    than zero, and each item is a number (not math.nan).
+
+    This returns a dictionary with key/value pairs meant for DB insertion,
+    top-level for BenchmarkResult.
+    """
+
+    # Encode invariants.
+    samples = stats_usergiven["data"]
+    assert len(samples) > 0
+    for sa in samples:
+        assert sa is not None
+        assert sa != math.nan
```

Comment:
Of course such code must go away. This should be locked in with proper type annotations. This was me thinking "OK, I _really_ work with these assumptions, and I am rather confident that they are met. And I really want to know when they are note met".

Reply:
For the record: `sa != math.nan` is an incorrect comparison method. Always use `math.isnan()`!